src/Apps/Order/Routes/Shipping2/index.tsx

```diff
@@ -69,203 +60,114 @@ export interface ShippingProps {
   commitMutation: CommitMutation
 }
 
-export type ShippingStage = "fulfillment_details" | "shipping_quotes"
-
-interface State {
-  newSavedAddressId: string | null
-  selectedShippingQuoteId: string | null
-  stage: ShippingStage
-}
-
 export const ShippingRoute: FC<ShippingProps> = props => {
   const { order, isCommittingMutation } = props
 
-  const saveFulfillmentDetails = useSaveFulfillmentDetails()
   const createSavedAddress = useCreateSavedAddress()
   const updateSavedAddress = useUpdateSavedAddress()
   const deleteSavedAddress = useDeleteSavedAddress()
   const selectShippingQuote = useSelectShippingQuote()
 
-  const orderContext = useComputeShippingContext(props)
+  const shippingContext = useComputeShippingContext(props)
 
   const {
     parsedOrderData,
     helpers: { fulfillmentDetails: fulfillmentFormHelpers },
-  } = orderContext
+  } = shippingContext
 
   const isOffer = order.mode === "OFFER"
   const isArtsyShipping =
     parsedOrderData.savedFulfillmentDetails?.isArtsyShipping
 
-  const [state, dispatch] = useReducer(shippingStateReducer, {
-    newSavedAddressId: null,
-    selectedShippingQuoteId: parsedOrderData.selectedShippingQuoteId ?? null,
-    stage: parsedOrderData.savedFulfillmentDetails?.isArtsyShipping
-      ? "shipping_quotes"
-      : "fulfillment_details",
-  })
-
-  const { requiresArtsyShippingTo, shippingQuotes } = parsedOrderData
-
   const advanceToPayment = useCallback(() => {
     props.router.push(`/orders/${props.order.internalID}/payment`)
   }, [props.router, props.order.internalID])
 
+  const handleFulfillmentDetailsSaved = useCallback(
+    ({ requiresArtsyShipping }: { requiresArtsyShipping: boolean }) => {
+      if (requiresArtsyShipping) {
+        shippingContext.helpers.setStage("shipping_quotes")
+      } else {
+        advanceToPayment()
+      }
+    },
+    [advanceToPayment, shippingContext.helpers]
+  )
```

Comment:
This is a new Prop for fulfillment details, along with `processUserAddressUpdates`. The diff in the rest of this file is big (much moved out) but the tests pass and the route works better than ever.